config_defaults_inc.php

```diff
@@ -2038,60 +2038,89 @@
 
 
 /**
- * Specifies the LDAP or Active Directory server to connect to.
+ * Specifies the LDAP or Active Directory server(s) to connect to.
+ * Multiple servers can be specified as a space-separated list.
  *
  * This must be a full LDAP URI (ldap[s]://hostname:port)
- * - Protocol can be either ldap or ldaps (for SSL encryption). If omitted,
- *   then an unencrypted connection will be established on port 389.
+ * - Protocol must be either:
+ *   - ldap (unencrypted or opportunistic StartTLS)
+ *   - ldaps (for TLS encryption)
  * - Port number is optional, and defaults to 389. If this doesn't work, try
  *   using one of the following standard port numbers: 636 (ldaps); for Active
  *   Directory Global Catalog forest-wide search, use 3268 (ldap) or 3269 (ldaps)
  *
  * Examples of valid URI:
- *   ldap.example.com
  *   ldap://ldap.example.com
  *   ldaps://ldap.example.com:3269/
  *
  * @global string $g_ldap_server
  */
-$g_ldap_server = 'ldaps://ldap.example.com/';
+$g_ldap_server = 'ldap://ldap.example.com/';
 
 /**
- * The root distinguished name for LDAP searches
+ * Decides if the connection will attempt an opportunistic upgrade to
+ * a TLS connection (STARTTLS).
+ * Be sure to use the ldap:// scheme and not ldaps://
+ * if setting this to true.  For security, a failure aborts the entire connection.
+ *
+ * @global bool $g_ldap_use_starttls
+ */
+$g_ldap_use_starttls = true;
+
+/**
+ * The minimum version of the TLS protocol to allow.
+ * For example, LDAP_OPT_X_TLS_PROTOCOL_TLS1_2. If OFF, then the protocol version
+ * is not set. This maps to the LDAP_OPT_X_TLS_PROTOCOL_MIN ldap library option.
+ * Requires PHP 7.1 or later. For security, a failure aborts the entire connection.
+ * @see https://www.php.net/manual/en/ldap.constants.php#constant.ldap-opt-x-tls-protocol-min
+ *
+ * @global int $ldap_tls_protocol_min
+ */
+$ldap_tls_protocol_min = OFF;
+
+/**
+ * The root distinguished name for LDAP searches.
  * @global string $g_ldap_root_dn
  */
 $g_ldap_root_dn = 'dc=example,dc=com';
 
 /**
- * LDAP search filter for the organization
+ * LDAP search filter for the organization.
  * e.g. '(organizationname=*Traffic)'
  * @global string $g_ldap_organization
  */
 $g_ldap_organization = '';
 
 /**
- * The LDAP Protocol Version, if 0, then the protocol version is not set.
- * For Active Directory use version 3.
+ * The LDAP Protocol Version.
+ * If 0, then the protocol version is not set, and you get whatever default
+ * the underlying ldap library uses. This maps to the LDAP_OPT_PROTOCOL_VERSION
+ * ldap library option. In almost all cases you should use 3. LDAPv3 was
+ * introduced back in 1997. LDAPv2 was deprecated in 2003 by RFC3494.
  *
  * @global integer $g_ldap_protocol_version
  */
-$g_ldap_protocol_version = 0;
+$g_ldap_protocol_version = 3;
```

Comment:
Changing the default might introduce regressions.

@dregad you insisted in not changing the default for `ldap_follow_referrals`.
Shouldn't we do the same for `ldap_protocol_version`?

Reply:
I think it makes sense to switch to 3, considering the issues we've had with accented characters in the past, e.g. 
- https://mantisbt.org/bugs/view.php?id=20809
- https://mantisbt.org/bugs/view.php?id=22655
- https://mantisbt.org/forums/viewtopic.php?f=3&t=26735

Reply:
And v3 is 24 years old!  Surely vastly more people are using v3 than v2.  Defaulting to v2 is just one more thing new users have to find and change before their LDAP works.